src/core/activityContent/validation.ts

```diff
@@ -40,386 +42,398 @@ const SUPPORTED_IMAGE_MEDIA_TYPES = [
+export const requireValidActivityContentNote = (obj: unknown): boolean => {
+  if (requireIsActivityContentNoteType(obj)) {
+    if (obj["tag"]) requireIsActivityContentTag(obj["tag"]);
+
+    if (obj["published"] && !isValidPublishedField(obj["published"]))
+      throw new InvalidActivityContentError("ActivityContentNote published is invalid");
+    if (obj["attachment"]) {
+      const attachments = requireToArray(obj["attachment"], "ActivityContentNote attachment");
+      requireGetSupportedContentAttachments(attachments);
     }
+    if (obj["location"]) requireIsActivityContentLocation(obj["location"]);
+  } else {
+    throw new InvalidActivityContentError("ActivityContentNote is invalid for an unknown reason");
   }
-  if (obj["location"]) {
-    if (Array.isArray(obj["location"])) {
-      for (const location of obj["location"]) {
-        if (!isValidActivityContentLocation(location)) return false;
-      }
-    } else {
-      if (!isValidActivityContentLocation(obj["location"])) return false;
-    }
-  }
-
   return true;
 };
 
 /**
- * isValidActivityContentProfile() validates that a given object meets all
+ * requireValidActivityContentProfile() validates that a given object meets all
  * necessary specification requirements for an ActivityContentProfile object,
  * include string field formats and other logical validations. Note that this
  * function **does not** attempt to fetch any icons to validate that their
  * hashes are correct.
  *
  * @param obj - The object to check
- * @returns A boolean indicating whether or not the object is a valid ActivityContentProfile
+ * @returns true if valid.
+ * @throws InvalidActivityContentError if not valid, with a message
  */
-export const isValidActivityContentProfile = (obj: unknown): boolean => {
-  if (!isActivityContentProfile(obj)) return false;
-
-  if (obj["published"] && !isValidPublishedField(obj["published"])) return false;
-  if (obj["icon"]) {
-    for (const icon of obj["icon"]) {
-      if (!isValidActivityContentImageLink(icon)) return false;
-    }
-  }
-  if (obj["location"]) {
-    if (Array.isArray(obj["location"])) {
-      for (const location of obj["location"]) {
-        if (!isValidActivityContentLocation(location)) return false;
+export const requireValidActivityContentProfile = (obj: unknown): boolean => {
+  if (requireIsActivityContentProfileType(obj)) {
+    if (obj["tag"] && !isArrayOfType(obj["tag"], requireIsActivityContentTag)) requireIsActivityContentTag(obj["tag"]);
+
+    if (obj["published"] && !isValidPublishedField(obj["published"]))
+      throw new InvalidActivityContentError("ActivityContentProfile published is invalid");
+    if (obj["icon"]) {
+      for (const icon of obj["icon"]) {
+        requireToActivityContentLink(icon);
       }
-    } else {
-      if (!isValidActivityContentLocation(obj["location"])) return false;
     }
+    if (obj["location"]) requireIsActivityContentLocation(obj["location"]);
```

Comment:
is it no longer valid for a profile to have multiple locations? That should be mentioned in changelog if so.